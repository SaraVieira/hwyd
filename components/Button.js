/* @flow */

import { TouchableOpacity, Text } from 'react-native';
import styled from 'styled-components';

const DefaultButton = styled(TouchableOpacity)`
  align-items: center;
  background-color: ${props => props.theme.colors.oldGeranium};
  border-radius: 3px;
  justify-content: center;
  padding: 10px 18px;
`;

const DefaultText = styled(Text)`
  color: ${props => props.theme.colors.white};
  font-family: ${props => props.theme.fonts.title};
`;

const OutlineButton = styled(TouchableOpacity)`
  align-items: center;
  border-color: ${props => props.theme.colors.oldGeranium};
  border-radius: 3px;
  border-width: 2px;
  justify-content: center;
  padding: 8px 16px;
`;

const OutlineText = styled(Text)`
  color: ${props => props.theme.colors.oldGeranium};
  font-family: ${props => props.theme.fonts.title};
`;

const TransparentButton = styled(TouchableOpacity)`
  align-items: center;
  border-radius: 3px;
  justify-content: center;
  padding: 10px 18px;
`;

const TransparentText = styled(Text)`
  color: ${props => props.theme.colors.oldGeranium};
  font-family: ${props => props.theme.fonts.title};
`;

export const Button = {
  Default: DefaultButton,
  Outline: OutlineButton,
  Transparent: TransparentButton,
};

export const ButtonText = {
  Transparent: TransparentText,
  Default: DefaultText,
  Outline: OutlineText,
};
