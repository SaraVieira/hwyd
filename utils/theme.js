import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
