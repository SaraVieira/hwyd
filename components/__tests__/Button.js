import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Button, ButtonText } from '../Button';
import theme from '../../utils/theme';

it('renders a transparent button (no background color)', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button.Transparent />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
