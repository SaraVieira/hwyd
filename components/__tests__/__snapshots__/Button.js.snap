// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a transparent button (no background color) 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px 18px;
}

<TouchableOpacity
  activeOpacity={0.2}
  className="c0"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Press Me
  </Text>
</TouchableOpacity>
`;
